packages-content-model/roosterjs-content-model-editor/lib/modelApi/common/mergeModel.ts

```diff
@@ -120,14 +120,21 @@ export function mergeModel(
 function mergeParagraph(
     markerPosition: InsertPoint,
     newPara: ContentModelParagraph,
-    mergeToCurrentParagraph: boolean
+    mergeToCurrentParagraph: boolean,
+    option: MergeModelOption | undefined
 ) {
     const { paragraph, marker } = markerPosition;
     const newParagraph = mergeToCurrentParagraph
         ? paragraph
         : splitParagraph(markerPosition, newPara.format);
     const segmentIndex = newParagraph.segments.indexOf(marker);
 
+    if (option?.mergeFormat == 'none' && mergeToCurrentParagraph) {
+        newParagraph.segments.forEach(segment => {
+            segment.format = { ...(newParagraph.segmentFormat || {}), ...segment.format };
```

Comment:
Are we passing the paragraph format to segment and them removing the paragraph format, right? If some segment already has a format that is different from the paragraph format, will the segment lose the format and apply only the paragraph format?

Reply:
Right, it should not. since the segment.format object we are destructing to create the new format would override the newParagraph.segmentFormat. If the format keys are duplicated. But maybe need to update this due to Jason's comment.